libraries/AP_Motors/AP_MotorsHeli_Swash.h

```diff
@@ -0,0 +1,88 @@
+/// @file	AP_MotorsHeli_Swash.h
+/// @brief	Swashplate Library for traditional heli
+#pragma once
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>            // ArduPilot Mega Vector/Matrix math Library
+#include <AP_Param/AP_Param.h>
+
+// swashplate types
+enum SwashPlateType {
+    SWASHPLATE_TYPE_H3 = 0,
+    SWASHPLATE_TYPE_H1,
+    SWASHPLATE_TYPE_H3_140,
+    SWASHPLATE_TYPE_H3_120,
+    SWASHPLATE_TYPE_H4_90,
+    SWASHPLATE_TYPE_H4_45
+};
+
+// collective direction
+enum CollectiveDirection {
+    COLLECTIVE_DIRECTION_NORMAL = 0,
+    COLLECTIVE_DIRECTION_REVERSED
+};
+
+class AP_MotorsHeli_Swash {
+public:
+    friend class AP_MotorsHeli_Single;
+    friend class AP_MotorsHeli_Dual;
+
+    AP_MotorsHeli_Swash() {};
+
+    // calculate_roll_pitch_collective_factors - calculate factors based on swash type and servo position
+    void calculate_roll_pitch_collective_factors();
+
+    // set_swash_type - sets swashplate type
+    void set_swash_type(SwashPlateType swash_type) { _swash_type = swash_type; }
+
+    // set_collective_direction - sets swashplate collective direction
+    void set_collective_direction(CollectiveDirection collective_direction) { _collective_direction = collective_direction; }
+
+    // set_phase_angle - sets swashplate phase angle
+    void set_phase_angle(int16_t phase_angle) { _phase_angle = phase_angle; }
+
+    // set_phase_angle - sets swashplate phase angle
+    float get_servo_out(int8_t servo_num, float pitch, float roll, float collective);
+
+    void set_servo1_pos(int16_t servo_pos) { _servo1_pos = servo_pos; }
```

Comment:
I guess these set_servoX_pos() functions are just to pass the parameter values into the Swash library.  I wonder instead if the parameters could be exposed more directly to AP_MotorsHeli_Swash class.

Reply:
the get_linear_servo_output(), set_linear_servo_out() functions and _make_servo_linear variable are also only to provide access to the _linear_swash_servo parameter.. just seems like we could handle this in another more direct way.